### PR TITLE
cookiecutter: modified files pt1

### PR DESCRIPTION
**Pull request by cadenmyers13**
@sbillinge splitting up the modified files PR

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -23,31 +23,32 @@ jobs:
...
+      - name: Validate diffpy.pdfgui
+        shell: bash -l {0}
         run: |
-          source $HOME/mambaforge/etc/profile.d/conda.sh
-          conda activate test_env
-          mamba install --file requirements/run.txt
-          mamba install --file requirements/test.txt
-          pip install -e .
-
-      - name: Use Xvfb Action
```
Comment (sbillinge) [inline .github/workflows/main.yml:49]: we will need this xvfb action for the wx.   

This PR is getting the tests to pass and will take some work, so put all the workflow work on a separate PR.   I had it working before with the mamba stuff and we may need to do it that way, but let's try and get as far as we can with the cookiecutter version, then see if we can the xvfb action to work.  WHen I was doing it before, it couldn't activate the conda environment for some reason.  Let's work on this on a separate PR.

**Review by sbillinge — COMMENTED** [review]
please see inline comments.

**Comment by Sparks29032** [comment]
@cadenmyers13 Please merge the PR I sent in before making edits on this branch. These are to fix the deprecations and modify the way we access the date of the package. Yes, it currently uses `os` rather than `Pathlib`, but I'm sure the migration should be easy, and I want to get this fix in before my computer blue-screens again. One of my RAM drivers is broken.

**Comment by Sparks29032** [comment]
The PR migrates `pkg_resources` and modifies how the `daemon` is accessed.

**Comment by Sparks29032** [comment]
Add `importlib_resources` to `build.txt` or `run.txt`. If you add it to `build.txt` you need to add the line `conda install --file requirements/build.txt` in the `main.yml`.

**Review by sbillinge — COMMENTED** [review]
@cadenmyers13 @Sparks29032 please let me know what to do with this.  Merge?  or wait.

**Comment by Sparks29032** [comment]
Let's try getting tests passing on the CI before merging. Hard :(

**Comment by cadenmyers13** [comment]
@sbillinge @Sparks29032 since `wxpython` requires python<=3.11 I changed the required python to be less than 3.12. The CI error now is saying that diffpy.pdfgui requires 3.12.4. Ideas on how to fix this?

**Review by sbillinge — COMMENTED** [review]
@cadenmyers13 let me know what you want me to do with this PR.  Are we working on getting the tests to pass?  Do we want to merge some things first and then build a new PR for trouble-shooting the tests?  That might seem like a better idea?

**Comment by cadenmyers13** [comment]
> @cadenmyers13 let me know what you want me to do with this PR.  Are we working on getting the tests to pass?  Do we want to merge some things first and then build a new PR for trouble-shooting the tests?  That might seem like a better idea?

I agree. Let's get this merged first then start on the CI fix.

**Comment by sbillinge** [comment]
@cadenmyers13 I'm closing this.  Please can you make a clean PR with everything not related to getting the code to run, so just .codecov.yml, .coveragerc, .gitattributes, .gitignore, CHANGELOG.rst, pyproject.toml README.rst then we will go more slowly with the rest.

**Pull request closed without merging**